npm/extensions.bzl

```diff
@@ -237,11 +264,28 @@ def _npm_import_attrs():
 
     return attrs
 
+def _exclude_package_contents_attrs():
+    return {
+        "packages": attr.string_list(
+            doc = "List of package names to apply exclusions to. Supports wildcards like '*' for all packages.",
+            default = [],
+        ),
+        "patterns": attr.string_list(
+            doc = "List of glob patterns to exclude from the specified packages.",
+            default = [],
+        ),
+        "use_defaults": attr.bool(
+            doc = "Whether to use default exclusion patterns for the specified packages.",
+            default = False,
+        ),
+    }
+
 npm = module_extension(
     implementation = _npm_extension_impl,
     tag_classes = {
         "npm_translate_lock": tag_class(attrs = _npm_translate_lock_attrs()),
         "npm_import": tag_class(attrs = _npm_import_attrs()),
+        "exclude_package_contents": tag_class(attrs = _exclude_package_contents_attrs()),
```

Comment:
I would vote to put the `npm_` prefix just to be consistent. We can debate dropping them all in v3 maybe?